src/components/xeniumMetrics/MetricsReader.tsx

```diff
@@ -0,0 +1,256 @@
+import React from 'react';
+import { Input } from '../forms/Input';
+import BlueButton from '../buttons/BlueButton';
+import LoadingSpinner from '../icons/LoadingSpinner';
+import FileIcon from '../icons/FileIcon';
+import FailIcon from '../icons/FailIcon';
+import UploadIcon from '../icons/UploadIcon';
+import PassIcon from '../icons/PassIcon';
+import { useFormikContext } from 'formik';
+import { Metric, XeniumMetricsForm } from '../../pages/XeniumMetrics';
+
+interface FileParserProps {
+  rowIndex: number;
+}
+
+/**
+ * Parses CSV data into an array of objects where each object represents a row in the CSV file.
+ * The keys of each object are column names (extracted from the CSV header), and the values
+ * are cell values from the corresponding row.
+ * are cell values from the corresponding row.
+ *
+ * @param csvData The CSV data to parse as a string.
+ * @returns An array of objects representing rows in the CSV file, with keys as column names
+ * and values as cell values.
+ */
+const parseCSV = (csvData: string): { [key: string]: string }[] => {
```

Comment:
Is there not an existing JS library that can handle parsing CSV? (and might also correctly check for quoted values)

Reply:
nothing that I am aware off, quoted values should be handled okay ( the quotes are considered as part of the value)

Reply:
What I mean by "correctly check for quoted values" is that separator characters inside a quoted value are typically not interpreted as separators. So the line
```
Alpha, Beta, "Gam,ma"
```
would be parsed as three values: `Alpha`, `Beta` and `Gam,ma`.
It could be that this will never come up, and the values in metrics will never contain quotes or commas; but an existing JS library might just handle these things correctly so we wouldn't have to worry about it.